src/datalink/mod.rs

```diff
@@ -68,7 +68,7 @@ pub fn datalink_channel(network_interface: &NetworkInterface,
     -> io::Result<(DataLinkSender, DataLinkReceiver)> {
     match backend::datalink_channel(network_interface, write_buffer_size, read_buffer_size,
                                              channel_type) {
-        Ok((tx, rx)) => Ok((DataLinkSender { dlsi: tx }, DataLinkReceiver { dlri: rx })),
+        Ok((sender, receiver)) => Ok((DataLinkSender { backend: sender}, DataLinkReceiver { backend: receiver })),
```

Comment:
Looks like I have found a companion. :-)
I am also that strict when it comes to code style.